sklearn/feature_selection/tests/test_feature_select.py

```diff
@@ -71,6 +73,27 @@ def test_f_classif():
     assert_array_almost_equal(pv_sparse, pv)
 
 
+@pytest.mark.parametrize("coeff", [abs_r_regression, r_regression])
+def test_r_regression(coeff):
```

Comment:
I would split the test to be more granular:

* one test to check the consistence between sparse and not sparse
* one that check the centering (this could also use `sklearn.testing._convert_container) to test on dense and sparse array
* and one that check the absolute equivalence

Reply:
Do we want to check that our implementation is close to the `np.corrcoef` from numpy?